format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSink.java

```diff
@@ -135,12 +127,28 @@ public void prepareRun(BatchSinkContext context) throws Exception {
     Map<String, String> outputProperties = new HashMap<>(validatingOutputFormat.getOutputFormatConfiguration());
     outputProperties.putAll(getFileSystemProperties(context));
     outputProperties.put(FileOutputFormat.OUTDIR, getOutputDir(context.getLogicalStartTime()));
-
     context.addOutput(Output.of(config.getReferenceName(),
                                 new SinkOutputFormatProvider(validatingOutputFormat.getOutputFormatClassName(),
                                                              outputProperties)));
   }
 
+  public ValidatingOutputFormat getOutputFormatForRun(BatchSinkContext context) throws InstantiationException {
```

Comment:
I assume this is pulled into its own method so subclasses can use it? If that's the case, can this be protected instead of public?

Reply:
Yes, protected is enough. Fixed.